tools/squidclient/squidclient.cc

```diff
@@ -425,17 +425,15 @@ main(int argc, char *argv[])
     }
 #endif
     /* Build the HTTP request */
+    const char *pathPassword = nullptr;
     if (strncmp(url, "mgr:", 4) == 0) {
         char *t = xstrdup(url + 4);
-        const char *at = nullptr;
-        if (!strrchr(t, '@')) { // ignore any -w password if @ is explicit already.
-            at = ProxyAuthorization.password;
+        // XXX: Bail on snprintf() failures
+        snprintf(url, sizeof(url), "http://%s:%hu/squid-internal-mgr/%s", Transport::Config.hostname, Transport::Config.port, t);
+        if (const auto at = strrchr(url, '@')) {
+            *at = 0; // send password in Proxy-Authorization header, not URL
+            pathPassword = at + 1; // embedded @password overwrites -w password further below
```

Comment:
@eduard-bagdasaryan, when you fix that bug in the official code (per Amos request), please mention the fix/change in in the PR description.

Reply:
Or, better, fix it in a dedicated PR! That dedicated PR will become a prerequisite for this PR (and all the other PRs you end up forking from this PR to address another change request).

Reply:
>  better, fix it in a dedicated PR!

I decided to fix this bug in this PR because these code blocks (including the new pathPassword variable) are changed  here anyway.